### DISCUSSION TITLE
Ignoring `jwks_uri` and injecting KeyStore instead.

### DISCUSSION THREAD

**Discussion by douglance** [panva/openid-client#345] [Q&A] [answered]
I edited the code to skip over the `jwks_uri` and use a KeyStore that is injected instead. And it seems to be working just as well. 

My issuer doesn't use the `jwks_uri`, so I need to be able to inject the KeyStore rather than fetching it. 

Is there a way to do that without forking the library?

**Comment by panva** [comment, marked as answer]
> Is there a way to do that without forking the library?

There is not. `jwks_uri` is a baseline requirement for this library whenever the client is meant to validate an assertion signed by the IdP using an asymmetric key. An OP without jwks_uri cannot pass any of the OIDC conformance profiles and is therefore out of scope for this library.
